src/react/src/deckgl.js

```diff
@@ -193,15 +198,17 @@ export default class DeckGL extends React.Component {
     // using the view descriptors
     const children = this._renderChildrenUnderViews(this.children);
 
-    // Render deck.gl as last child
-    const {id, width, height, style} = this.props;
-    const deck = createElement('canvas', {
-      ref: c => (this.overlay = c),
-      key: 'overlay',
+    // Note that width and height are handled by deck.gl
+    const {id, style} = this.props;
+    const canvas = createElement('canvas', {
+      ref: c => (this.deckCanvas = c),
+      key: 'deck-canvas',
       id,
-      style: Object.assign({}, {position: 'absolute', left: 0, top: 0, width, height}, style)
+      style: Object.assign({}, {position: 'absolute', left: 0, top: 0}, style)
```

Comment:
Doesn't this overwrite the style set in `Deck`?

Reply:
`style` and `width` `height` can potentially conflict with each other. My suggestion is to still require `width` and `height` be numbers, but make them optional. CSS-style dimensions can be applied via the `style` object.